addons/portal/views/portal_templates.xml

```diff
@@ -504,7 +504,7 @@
             <select name="state_id" t-attf-class="form-select #{error.get('state_id') and 'is-invalid' or ''}">
                 <option value="">select...</option>
                 <t t-foreach="states or []" t-as="state">
-                    <option t-att-value="state.id" style="display:none;" t-att-data-country_id="state.country_id.id" t-att-selected="state.id == int(state_id) if state_id else state.id == partner.state_id.id">
+                    <option t-att-value="state.id" style="" t-att-data-country_id="state.country_id.id" t-att-selected="state.id == int(state_id) if state_id else state.id == partner.state_id.id">
```

Comment:
The reason it's d-none by default is that we only unhide states from the currently-selected country in js. See `_adaptAddressForm`.

If it doesn't work, it's probably not getting countryId correctly. I can't replicate with the steps you've shown.
I see "the same problem isn't in runbot", so is it an issue with a custom view on their side?